packages/dnb-eufemia/src/components/upload/types.ts

```diff
@@ -16,7 +16,12 @@ export type UploadProps = {
   /**
    * unique id used with the useUpload hook to manage the files
    */
-  id: string
+  id:
+    | string
+    | (() => void)
+    | Promise<() => void>
+    | React.Context<any>
+    | Record<string, unknown>
```

Comment:
```suggestion
  id: SharedStateId
```

NB: The import is missing.

Reply:
Yeah, of course. 
Not sure why I did not do it, I think I thought it was from Forms, but it's clearly not 👍

Reply:
Updated now :)